engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
+    val namespace = validateNamespace(wdlSource)
+    val result = namespace match {
+      case scalaz.Success(ns) =>
+        val options = validateWorkflowOptions(CromwellBackend.backend(), workflowOptions: Option[WdlJson])
+        val inputs = validateInputs(ns, workflowInputs)
+        val runtimeAttributes = validateRuntimeAttributes(ns)
+        (options |@| inputs |@| runtimeAttributes) ((opt, in, runAttr) => ValidationSuccess(ns, Option(in), Option(opt), runAttr))
+
+      case scalaz.Failure(failure) => failure.failure
+    }
+
+    result match {
+      case scalaz.Success(success) => success
+      case scalaz.Failure(failure) =>
+        ValidationFailure(new ValidationException(s"Workflow input processing failed.", failure))
     }
   }
 
-  // TODO: With PBE, this should be defined in the backend.
-  private def validateRuntimeAttributes(namespaceWithWorkflow: NamespaceWithWorkflow): Seq[Set[String]] = {
-    TryUtil.sequence(namespaceWithWorkflow.workflow.calls map {
-      call => CromwellRuntimeAttributes.validateKeys(call.task.runtimeAttributes.attrs.keySet, CromwellBackend.backend().backendType)
-    }) match {
-      case Success(validatedRuntimeAttrs) => validatedRuntimeAttrs
-      case Failure(reason) => throw new IllegalArgumentException("Failed to validate runtime attributes.", reason)
+  private def validateNamespace(source: WdlSource): ErrorOr[NamespaceWithWorkflow] = {
+    try {
+      NamespaceWithWorkflow.load(source).successNel
+    } catch {
+      case e: Exception => s"Unable to load namespace from workflow: ${e.getMessage}".failureNel
     }
   }
 
-  private def validateInputs(workflowInputs: Option[WdlJson], namespaceWithWorkflow: NamespaceWithWorkflow): Option[WorkflowCoercedInputs] = {
-    def rawInputs(json: WdlJson): Map[String, JsValue] = {
+  private def validateInputs(namespaceWithWorkflow: NamespaceWithWorkflow, workflowInputs: Option[WdlJson]): ErrorOr[WorkflowCoercedInputs] = {
+    def validateRawInputs(json: WdlJson): ErrorOr[Map[String, JsValue]] = {
       Try(json.parseJson) match {
-        case Success(JsObject(inputs)) => inputs
-        case Failure(reason: Throwable) => throw new Throwable(reason.getMessage)
-        case _ => throw new IllegalArgumentException(s"Failed to parse the workflow inputs JSON: $json")
+        case Success(JsObject(inputs)) => inputs.successNel
+        case Failure(reason: Throwable) => s"Workflow contains bad inputs JSON: ${reason.getMessage}".failureNel
+        case _ => s"Workflow contains bad inputs JSON: $json".failureNel
```

Comment:
Since you've broken this out into a separate case, this message should be a bit more detailed.  Technically a bare string is valid JSON but not a valid `JsObject`.